### PR TITLE
update with social links and careers section

### PR DESCRIPTION
**Pull request by nwhite1**
## Description

just some changes to the readme file with social links and careers section.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,3 +66,6 @@ According to SemVer, anything in the public API can change at any time before ve
 
 For more information on Tendermint Core and pointers to documentation for Tendermint visit
 this [repository](https://github.com/tendermint/tendermint).
+
+### Careers
+We are hiring Go engineers! Join us in building the future of blockchain scaling and interoperability. [Apply here](https://angel.co/company/celestialabs/jobs).
```
Comment (liamsi) [inline README.md:71]: I'd either suggest moving the Careers section above the Resources one, or, alternatively make it not a sub-section of Resources but one of the parent section. The latter could be achieved by:
```suggestion
## Careers
We are hiring Go engineers! Join us in building the future of blockchain scaling and interoperability. [Apply here](https://angel.co/company/celestialabs/jobs).
```

**Comment by liamsi** [comment]
Rendered: https://github.com/nwhite1/celestia-core/tree/patch-1#readme

**Review by evan-forbes — APPROVED** [review]
thanks @nwhite1 !

**Review by liamsi — APPROVED** [review]
Dope, thanks a lot @nwhite1 🚀

**Comment by nwhite1** [comment]
wooo first PR merged! 🚀

**Pull request merged**